### DISCUSSION TITLE
Does the vscode-cpptools extension currently support C++20 modules?

### DISCUSSION THREAD

**Discussion by bowenxuuu** [microsoft/vscode-cpptools#12501] [Q&A] [answered]
`C++20 Modules` is a great feature, and it's well liked by many. I'm developing a program leveraging C++20 Modules in VSCode, and my code both compiles and runs without any issues. However, the `cpptools` extension always throws numerous error messages at me, suggesting it can't locate the modules I've defined. My project is built using `CMake`, and I'm using the latest versions of `VSCode`, `Visual Studio`, `MSVC`, and the `vscode-cpptools` extension. I'm not sure what's going wrong, as my program compiles and runs perfectly, yet the `vscode-cpptools` extension consistently flags errors.
﻿
I wonder if the `vscode-cpptools` plugin now supports the `C++20 Modules` feature? It would be helpful if someone could provide a minimal example of code using the `C++20 Modules` feature (based on `CMake`).

**Comment by sean-mcmanus** [comment, marked as answer]
@xb284524239 Modules support is still in development, so it has bugs, such as https://github.com/microsoft/vscode-cpptools/issues/9965 .

**Reply by bowenxuuu** [reply]
Hearing this news in 2024 is disappointing!

**Reply by hxznhf** [reply]
It's September 2025 now, and this issue still hasn't been resolved. It's hard to imagine that even though the C++20 standard has been out for so many years, the support from relevant tools is still experimental.
